clientconn.go

```diff
@@ -1589,24 +1589,8 @@ func parseTarget(target string) (resolver.Target, error) {
 	if err != nil {
 		return resolver.Target{}, err
 	}
-	// For targets of the form "[scheme]://[authority]/endpoint, the endpoint
-	// value returned from url.Parse() contains a leading "/". Although this is
-	// in accordance with RFC 3986, we do not want to break existing resolver
-	// implementations which expect the endpoint without the leading "/". So, we
-	// end up stripping the leading "/" here. But this will result in an
-	// incorrect parsing for something like "unix:///path/to/socket". Since we
-	// own the "unix" resolver, we can workaround in the unix resolver by using
-	// the `URL` field instead of the `Endpoint` field.
-	endpoint := u.Path
-	if endpoint == "" {
-		endpoint = u.Opaque
-	}
-	endpoint = strings.TrimPrefix(endpoint, "/")
 	return resolver.Target{
-		Scheme:    u.Scheme,
-		Authority: u.Host,
-		Endpoint:  endpoint,
```

Comment:
Just because these fields are marked as deprecated, does not mean that we can stop setting them. Marking them as deprecated is only a signal to our users to start using the new `URL` field. But, we cannot break existing users who are continuing to use the old deprecated fields. 

What https://github.com/grpc/grpc-go/issues/5754 asks is for internal usages of the deprecated fields to be replaced with the new field. It does not ask for not populating these old fields.

Reply:
Sorry, I misunderstood.